api/filter_dimension_options.go

```diff
@@ -457,20 +451,31 @@ func (api *FilterAPI) patchFilterBlueprintDimensionHandler(w http.ResponseWriter
 	log.Event(ctx, "successfully patched filter dimension options on filter blueprint", log.INFO, logData)
 }
 
-// patchFilterBlueprintDimension applies the patches by calling add or remove filter dimension options. It keeps track of a list of successful patches, so that
-func (api *FilterAPI) patchFilterBlueprintDimension(ctx context.Context, filterBlueprint *models.Filter, dimensionName string, patches []dprequest.Patch, logData log.Data) (successful []dprequest.Patch, err error) {
+// patchFilterBlueprintDimension applies the patches by calling add or remove filter dimension options. It keeps track of a list of successful patches
+func (api *FilterAPI) patchFilterBlueprintDimension(ctx context.Context, filterBlueprintID string, dimensionName string, patches []dprequest.Patch, logData log.Data) (successful []dprequest.Patch, err error) {
 
 	successful = []dprequest.Patch{}
-
-	// if filter blueprint does not have dimension, return now with ErrDimensionNotFound, as no path operation can succeed
-	hasDimension, _, _ := findDimensionAndOptions(filterBlueprint, dimensionName, []string{})
-	if !hasDimension {
-		return successful, filters.ErrDimensionNotFound
-	}
+	dimensionValidated := false
 
 	// apply patch operations sequentially, stop processing if one patch fails, and return a list of successful patches operations
 	for _, patch := range patches {
 		options := removeDuplicateAndEmptyOptions(patch.Value)
+
+		// Get filter Blueprint in all iterations to prevent conflicts due to the timestamp check in mongoDB when adding or removing values
```

Comment:
This is a good point. I think for all updates we should always be updating the latest stored version.